src/analysis/complex/upper_half_plane/functions_bounded_at_infty.lean

```diff
@@ -0,0 +1,78 @@
+/-
+Copyright (c) 2022 Chris Birkbeck. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Birkbeck, David Loeffler
+-/
+
+import algebra.module.submodule.basic
+import analysis.complex.upper_half_plane.basic
+import order.filter.zero_and_bounded_at_filter
+
+/-!
+# Bounded at infinity
+
+For complex valued functions on the upper half plane, this file defines the filter `at_I_infty`
+required for defining when functions are bounded at infinity and zero at infinity.
+Both of which are relevant for defining modular forms.
+
+-/
+
+open complex filter
+
+open_locale topological_space upper_half_plane
+
+noncomputable theory
+
+namespace upper_half_plane
+
+/--Filter for approaching `i∞`-/
+def at_I_infty := filter.at_top.comap upper_half_plane.im
+
+lemma at_I_infty_basis : (at_I_infty).has_basis (λ _, true) (λ (i : ℝ), im ⁻¹' set.Ici i) :=
+filter.has_basis.comap upper_half_plane.im filter.at_top_basis
+
+lemma at_I_infty_mem (S : set ℍ) : S ∈ at_I_infty ↔ (∃ A : ℝ, ∀ z : ℍ, A ≤ im z → z ∈ S) :=
+begin
+  simp only [at_I_infty, filter.mem_comap', filter.mem_at_top_sets, ge_iff_le, set.mem_set_of_eq,
+    upper_half_plane.coe_im],
+  split,
+  { intro h, cases h with a h, refine ⟨a, (λ z hz, by {apply h (im z) hz , refl})⟩ },
+  { refine (λ h, by {cases h with A h,
+    refine ⟨A, (λ b hb x hx, by {apply (h x), rw hx, exact hb})⟩}) }
```

Comment:
I find the refine+by combination hard to read, especially with this indentation, so maybe
```suggestion
  { rintro ⟨A, h⟩,
    refine ⟨A, λ b hb x hx, h x _⟩,
    rwa hx, }
```
or perhaps less readable in different ways:
```suggestion
  { exact λ ⟨A, h⟩, ⟨A, λ b hb x hx, h x $ hb.trans_eq hx.symm⟩ },
```
```suggestion
  { exact Exists.imp (λ A h b hb x hx, h x $ hb.trans_eq hx.symm) },
```